ccdproc/image_collection.py

```diff
@@ -652,10 +663,17 @@ def _generator(self, return_type,
 
             file_name = path.basename(full_path)
 
+            if 'unit' not in ccd_kwargs:
```

Comment:
This shouldn't be needed, there is already some [code inside the reader](https://github.com/astropy/ccdproc/blob/master/ccdproc/ccddata.py#L643) to evaluate this. It's easier to maintain if the behaviour does not differ and is only implemented in one place.